packages/angular/projects/ui-angular/src/lib/components/authenticator/components/authenticator/authenticator.component.ts

```diff
@@ -61,12 +66,47 @@ export class AuthenticatorComponent
       formFields,
     } = this;
 
+    this.unsubscribeHub = listenToAuthHub((event) => {
+      /**
+       * Hub events aren't properly caught by Angular, because they are
+       * synchronous events. Angular tracks async network events and
+       * html events, but not synchronous events like hub.
+       *
+       * On any notable hub events, we run change detection manually.
+       */
+      const state = this.authenticator.authService.send(event);
+      this.changeDetector.detectChanges();
+
+      /**
+       * Hub events that we handle can lead to multiple state changes:
+       * e.g. `authenticated` -> `signOut` -> initialState.
+       *
+       * We want to ensure change detection runs all the way, until
+       * we reach back to the initial state. Setting the below flag
+       * to true to until we reach initial state.
+       */
+      this.isHandlingHubEvent = true;
+      return state;
```

Comment:
In other words -- on hub events we handle, we turn this flag on that will trigger manual change detection until state machine is back at the initial state (as result of a sign out).

This is based on current implementation of `listenToHub`, which only has `signOut` behavior or `refreshUser` behavior. 

This might change if we add more hub event handling. But this at least shouldn't be breaking when that happens; there just will be wasted change detections that we can look to optimize at then.

Reply:
This could happen with `tokenRefresh_failure`, because token expires silently instead of it firing off some html button press for example.